src/posts/2020-09-19-how-to-write-accessible-forms.md

```diff
@@ -0,0 +1,154 @@
+
+```css
+input:focus {
+  outline: 2px solid royalblue;
+  box-shadow: 1px 1px 8px 1px royalblue;
+}
+```
+
+## Break long forms into smaller sections
+
+Filling long forms can be stressful. This is largely true for almost everyone but particularity for people  with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html). Such forms can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) which isn't what anyone wants. A simple solution is to break them into smaller sections. This helps make long forms less daunting and easier to understand as it is provided in bits.
+
+The following principles apply for multi-step forms:
+
+- Repeat overall instruction on every page.
+- The forms should be split into logical groups.
+- The forms should inform the person about the progress thay are making.
+
+**Note**: There are no hard rules that defines a long form. However, you should consider if the form can be broken down into different logical sections.
+
+## Provide error messages
+
+When people encounter errors, they may need help to determine what's wrong and why. A common way forms indicate invalid fields is to change the `<input>` border color to red which is not ideal. The reason for this is because a change in color alone is not exposed to assistive technologies, so indicating errors in such ways will not inform the person of the error.
+
+The following are ways to effectively and accessibly communicate form errors:
+
+- Let the person know there is an error.
+- Let the person access the error, and provide instructions on how to fix it.
+- Allow the person re-submit the form.
+
+In the code below, there is an empty [live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) which report errors to assistive technologies. Additionally, each field is followed by a _hint_ which is hidden by default but shown if that field does not validate.
+
+```html
+<form>
+  <label for="username">Enter a username</label>
+  <input type="text" id="username">
+  <p class="hint hidden" id="usernameHint" >Your username cannot contain punctuation</p>
+
+  <label for="email">Enter your email address</label>
+  <input type="text" id="email">
+  <p class="hint hidden" id="emailHint">Please enter a valid email address</p>
+
+  <label for="password">Enter a password</label>
+  <input type="password" id="password">
+  <p class="hint hidden" id="passwordHint">Your password must be at least 6 characters</p>
+
+  <div aria-live="assertive" id="message"></div>
+ 
+  <button type="submit">Submit</button>
+</form>
+```
+
+```css
+.hidden {
+  display: none
+}
+
+.hint {
+  color: red;
+}
+```
+If validation errors occur, two things happen:
+
+1. The [live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) is populated upon form submission, announcing to assistive devices that there are errors.
+

```

Comment:
Removing this empty space will make the markdown parser render these two bullet points as an ordered list.